src/config.c

```diff
@@ -767,8 +865,15 @@ static int cgroup_config_create_groups(void)
 
 		error = cgroup_create_cgroup(cgroup, 0);
 		cgroup_dbg("creating group %s, error %d\n", cgroup->name, error);
-		if (error)
-			return error;
+		if (error) {
+			/*
+			 * Attempt to convert the controller version
+			 * and retry.
+			 */
+			error = convert_controller_versions(cgroup);
```

Comment:
How would this work on a hybrid system where some controllers are v1 and some are v2?   And do we care?

Reply:
It works for all the cases, please consider a hybrid cgroup setup, where the `cpu` controller is mounted as cgroup v2 controller:
```
# cat /proc/cmdline 
BOOT_IMAGE=/vmlinuz-6.2.0-rc2+ root=/dev/mapper/ubuntu--vg-ubuntu--lv ro maybe-ubiquity console=ttyS0,9600 console=ttyS1,9600 systemd.unified_cgroup_hierarchy=0 systemd.legacy_systemd_cgroup_controller=false cgroup_no_v1=cpu
# cgget -m
Hybrid mode (Cgroup v1/v2).

# cat cgconfig.conf
group a {
        cpu {
                cpu.shares = 756;
        }
}
```
```
# ./src/tools/cgget -r cpu.weight foo
foo:
cpu.weight: 100
```
```
# ./src/tools/cgconfigparser -l ./cgconfig.conf 
# ./src/tools/cgget -r cpu.weight foo
foo:
cpu.weight: 73
```